### DISCUSSION TITLE
[Feature Request] Generally improve the Performance of the final report

### DISCUSSION THREAD

**Discussion by DannyDainton** [DannyDainton/newman-reporter-htmlextra#361] [Ideas]
The reporter works quite well when the Collections being run are either small (Need to benchmark this) or the iteration count is low (under 500) - This becomes awful when the Collection contains thousands of Requests in hundreds of Folders all run thousands of times - Not sure who reads that report and all that data but users do that 😄 

The report will just get bigger and bigger and in turn, become super slow when initially opening the file and navigating between the different sections. There are a number of flags available to reduce the amount of data going into the report to take the weight out but it's just a band-aid covering the main issue.

To do things like syntax highlighting of the different pieces of code, it's making calls out to a 3rd party CDN and for large reports that just not scaleable. Something needs to be handled there.

There are other areas that can be identified but without out proper profiling of what's going on, I'm just guessing.

The templates are getting a bit clunky and a lot of the HTML/CSS/JS and handlebars stuff probably needs to be refactored.

**Comment by DannyDainton** [comment]
Using the `--reporter-htmlextra-noSyntaxHighlighting` flag can help reduce some of the issues with larger reports but it won't solve all the problems.

**Comment by poorna5500** [comment]
@DannyDainton,

Kindly do not close the issue, understand that linked to the main bug.. but the best practice is not close sub-bug too until and unless the main and related/sub-bug are closed just for the information sir.

Thanks,
Poorna.

**Comment by DannyDainton** [comment]
The issue you raised is for the same problem - There's no point in having multiple duplicate issues open for the same thing.

It's a general problem that impacts reports that contain lots of information.

**Comment by poorna5500** [comment]
Hi @DannyDainton, 
Ok, that makes sense. At the same time when can, we expect the resolution for this issue? Any idea..?

**Comment by DannyDainton** [comment]
Nope, it's not something I'm actively working on or will be anytime soon.

I've got far too much of my actual paid job's work to do before any of the issues in this repo. 😂

It's all open source so please feel free to take a look at how it could be improved, PRs are very welcome.

**Comment by poorna5500** [comment]
@DannyDainton, I completely understand the pain in the neck... haha.. sure will take a look at it, sir.

Thanks,
Poorna.

**Comment by ranjan1522** [comment]
> Nope, it's not something I'm actively working on or will be anytime soon.
> 
> I've got far too much of my actual paid job's work to do before any of the issues in this repo. 😂
> 
> It's all open source so please feel free to take a look at how it could be improved, PRs are very welcome.

any solution had been found?

**Comment by DannyDainton** [comment]
Nope, not worked on it. If you would like to take a look and offer any improvements, that would be welcome but I'm not currently doing anything in that area. 

You could use the `--reporter-htmlextra-noSyntaxHighlighting` flag to see if that helps or try reducing the report data down to what you need, using the other flags available.

**Comment by half-eaten-apple** [comment]
@DannyDainton After using '--reporter-htmlextra-noSyntaxHightlighting' flag also, i see lot of slowness in opening the html report. I have total 94 requests in my collection and 200+ assertions. html file size = 22342 kb.
Using htmlextra version 1.22.3

**Comment by DannyDainton** [comment]
It was a suggestion that it _might_ help with loading collections that have a huge request/response. I haven't done anything to the reporter in terms of performance so it's not going to get 'faster'.

You could try reducing the information in the final report, using the other flags to not show specific data.

If you have any suggestions or if you wanted to try and help here, all PRs are welcome.
